python/ql/lib/semmle/python/dataflow/new/internal/DataFlowDispatch.qll

```diff
@@ -1520,19 +1520,21 @@ abstract class ParameterNodeImpl extends Node {
 }
 
 /**
- * The value of a lambda itself at function entry, viewed as a node in a data
+ * The value of a closure itself at function entry, viewed as a node in a data
  * flow graph.
  *
  * This is used for tracking flow through captured variables, and we use a
  * separate node and parameter/argument positions in order to distinguish
  * "lambda self" from "normal self", as lambdas may also access outer `self`
  * variables (through variable capture).
  */
-class LambdaSelfReferenceNode extends ParameterNodeImpl, TLambdaSelfReferenceNode {
+class SynthCapturingClosureParameterNode extends ParameterNodeImpl,
```

Comment:
Thanks for listing to my feedback and updating the name :heart: 

I'm going to be annoying and question whether the new name is as good as it can get. :thinking: To my eyes the behavior we're using this for is to pass in values for the variables that have been captured. To me this new name doesn't directly translate to that behavior.

Once again, I will suggest `SynthCapturedVariablesParameter` / SynthCapturedVariablesArgument`. I guess you didn't fancy them for some reason, so maybe let's talk it over :blush: 

Maybe I'm bike-shedding too much. I'm wondering what it's called in the other languages?

Reply:
> To my eyes the behavior we're using this for is to pass in values for the variables that have been captured.

Yes, but as fields on this parameter. It represents the heap of the closure.

In Ruby it is called `LambdaSelfReferenceNode`.

Reply:
In Javascript, they chose `FunctionSelfReferenceNode`.
In Java it is `InstanceParameterNode`.

Reply:
I came round to `SynthCapturedVariablesParameterNode` in the end. I still read it as "a parameter that has something to do with captured variables" but I think that is actually fine. A name is only so long, and this gives the right association. I wrote a QLDoc with the precise interpretation instead, and it was much easier to write that than to try to cram it all into a name 👍 I just hope we will not need more synthetic parameters related to captured variables in the future 😁